curp/src/rpc/mod.rs

```diff
@@ -635,24 +635,36 @@ impl CurpError {
         Self::Internal(reason.into())
     }
 
-    /// Errors that should return early to the retry layer when we
-    /// got the error at propose stage
-    pub(crate) fn return_early(&self) -> bool {
-        matches!(
-            *self,
+    /// Get the priority of the error
+    pub(crate) fn priority(&self) -> CurpErrorPriority {
+        match *self {
             CurpError::Duplicated(_)
-                | CurpError::ShuttingDown(_)
-                | CurpError::InvalidConfig(_)
-                | CurpError::NodeAlreadyExists(_)
-                | CurpError::NodeNotExists(_)
-                | CurpError::LearnerNotCatchUp(_)
-                | CurpError::ExpiredClientId(_)
-                | CurpError::WrongClusterVersion(_)
-                | CurpError::Redirect(_)
-        )
+            | CurpError::ShuttingDown(_)
+            | CurpError::InvalidConfig(_)
+            | CurpError::NodeAlreadyExists(_)
+            | CurpError::NodeNotExists(_)
+            | CurpError::LearnerNotCatchUp(_)
+            | CurpError::ExpiredClientId(_)
+            | CurpError::Redirect(_) => CurpErrorPriority::ReturnImmediately,
+            CurpError::WrongClusterVersion(_) => CurpErrorPriority::High,
+            CurpError::RpcTransport(_) | CurpError::Internal(_) | CurpError::KeyConflict(_) => {
+                CurpErrorPriority::Low
+            }
+        }
     }
 }
 
+/// The priority of curp error, indicate which error should be handled in retry layer
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub(crate) enum CurpErrorPriority {
+    /// Low priority
+    Low = 1,
```

Comment:
Please add a comment to explain what should you do when encountering a low-priority `CurpError`? IMO, a better way is to rename the `Low` to something, like `ReturnImmediately`, a more self-explain name.

Reply:
Added some comments

Reply:
Both `High` and `ReturnImmediately` will cause the RPC return early, what's the difference between them?

Reply:
The client receiving a high-priority error does not necessarily mean the request cannot be completed, but receiving a ReturnImmediately error will trigger a direct retry.

Reply:
Ok, I get it. Please add this difference in the comment.

Reply:
Added